include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -947,46 +1002,57 @@ class operational_domain_impl
      */
     operational_status is_step_point_operational(const step_point& sp) noexcept
     {
-        // if the point has already been sampled, return the stored operational status
-        if (const auto op_value = has_already_been_sampled(sp); op_value.has_value())
+        static std::mutex mutex_to_protect_member_variables;
+
         {
-            return *op_value;
+            const std::lock_guard lock{mutex_to_protect_member_variables};
+            if (const auto op_value = has_already_been_sampled(sp); op_value.has_value())
+            {
+                return *op_value;
+            }
         }
 
-        // fetch the x and y dimension values
         const auto param_point = to_parameter_point(sp);
 
         const auto operational = [this, &param_point]()
         {
+            const std::lock_guard lock(mutex_to_protect_member_variables);
```

Comment:
@marcelwa As mentioned in our discussion, I had to add these mutex to avoid undefined behavior on my Mac machine.